lib/bamboo/adapters/mailgun_helper.ex

```diff
@@ -63,8 +73,19 @@ defmodule Bamboo.MailgunHelper do
       |> MailgunHelper.substitute_variables(%{ "greeting" => "Hello!", "password_reset_link" => "https://example.com/123" })
     
   """
-  def substitute_variables(email, variables = %{}) do
-    custom_vars = Map.get(email.private, :mailgun_custom_vars, %{})
-    Email.put_private(email, :mailgun_custom_vars, Map.merge(custom_vars, variables))
+  def substitute_variables(%Email{headers: headers} = email, variables = %{}) do
+    header_key = "X-Mailgun-Variables"
```

Comment:
How do you feel about promoting `"X-Mailgun-Variables"` to a module attribute? Perhaps something that expresses: `@mailgun_header_for_custom_vars` ?

Reply:
@stratigos Agreed. I added the module attribute as you suggested.